pandas/_libs/tslibs/timezones.pyx

```diff
@@ -103,7 +103,14 @@ cpdef inline tzinfo maybe_get_tz(object tz):
             if isinstance(tz, _dateutil_tzfile) and '.tar.gz' in tz._filename:
                 tz._filename = zone
         else:
-            tz = pytz.timezone(tz)
+            if tz.startswith('+') or tz.startswith('-') or tz.startswith('UTC+') or tz.startswith('UTC-'):
+                m = re.match(r'(?P<sign>[+-])(?P<hour>\d{2}):(?P<minutes>\d{2})', tz).groupdict()
+                if m['sign'] == '-':
+                    tz = pytz.FixedOffset(-60 * int(m['hour']) + int(m['minutes']))
+                else:
+                    tz = pytz.FixedOffset(60*int(m['hour']) + int(m['minutes']))
+            else:
+                tz = pytz.timezone(tz)
```

Comment:
im missing something; how is this connected to the tests?

Reply:
Pandas doesn't recognize timezones (`+02:00`, `UTC-04:00`) without a name (like `Europe/Lisbon`). This adds this ability. I've updated the PR so it doesn't need regex.

Reply:
but what pandas function are you referring to?  pd.Timestamp?  pd.to_datetime?

Reply:
See #35997 for details. This happened in two cases:

- When tried to serialize to parquet using fastparquet
- When tried to read parquet using pyarrow when the timezone aware date was an index

Timezone was an offset without a name in both cases.

Reply:
The test cases added are failing with Pandas 1.1.1, but this patch fixes the issue

Reply:
*almost:
```
_______________ TestParquetFastParquet.test_timezone_aware_index _______________
[gw0] linux -- Python 3.7.7 /home/travis/miniconda3/envs/pandas-dev/bin/python
self = <pandas.tests.io.test_parquet.TestParquetFastParquet object at 0x7f35438c6e90>
fp = 'fastparquet'
    def test_timezone_aware_index(self, fp):
        idx = [
            datetime.datetime.strptime(
                "2019-01-04T16:41:24+0200", "%Y-%m-%dT%H:%M:%S%z"
            )
        ]
        df = pd.DataFrame(index=idx)
>       check_round_trip(df, fp)
pandas/tests/io/test_parquet.py:867: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/tests/io/test_parquet.py:181: in check_round_trip
    compare(repeat)
pandas/tests/io/test_parquet.py:176: in compare
    expected, actual, check_names=check_names, check_like=check_like
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
l = DatetimeIndex(['2019-01-04 16:41:24+02:00'], dtype='datetime64[ns, UTC+02:00]', freq=None)
r = DatetimeIndex(['2019-01-04 16:41:24+02:00'], dtype='datetime64[ns, pytz.FixedOffset(120)]', name='index', freq=None)
obj = 'DataFrame.index'
    def _check_types(l, r, obj="Index"):
        if exact:
            assert_class_equal(l, r, exact=exact, obj=obj)
    
            # Skip exact dtype checking when `check_categorical` is False
            if check_categorical:
>               assert_attr_equal("dtype", l, r, obj=obj)
E               AssertionError: DataFrame.index are different
E               
E               Attribute "dtype" are different
E               [left]:  datetime64[ns, UTC+02:00]
E               [right]: datetime64[ns, pytz.FixedOffset(120)]```

Reply:
I have no idea for this one :/

Reply:
Changed `pytz.FixedOffset()` to `timezone(-timedelta(hours=hours, minutes=minutes))` and now the dtypes match too.

Reply:
@jbrockmendel does my explanation help?